packages/core-transaction-pool/src/wallet-manager.ts

```diff
@@ -24,60 +24,28 @@ export class WalletManager extends Wallets.WalletManager {
         this.forgetByAddress(Identities.Address.fromPublicKey(publicKey));
     }
 
-    public incrementNonce(publicKey: string): void {
-        this.findByPublicKey(publicKey).incrementNonce();
-    }
-
-    public decrementNonce(publicKey: string): void {
-        this.findByPublicKey(publicKey).decrementNonce();
-    }
-
-    public throwIfApplyingFails(transaction: Interfaces.ITransaction): void {
+    public senderIsKnownAndTrxCanBeApplied(transaction: Interfaces.ITransaction): void {
         // Edge case if sender is unknown and has no balance.
         // NOTE: Check is performed against the database wallet manager.
-        const { senderPublicKey } = transaction.data;
+        const senderPublicKey: string = transaction.data.senderPublicKey;
```

Comment:
Why remove the destructure?

Reply:
Because this way the reader can be hinted about the type of the variable. You know:
```
const x: string = someMethod();
```
is to be preferred over
```
const x = someMethod();
```